src/Debug/TraceablePublisher.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Debug;
+
+use Symfony\Component\Mercure\PublisherInterface;
+use Symfony\Component\Mercure\Update;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * Traces updates for profiler.
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ *
+ * @experimental
+ */
+final class TraceablePublisher implements PublisherInterface
```

Comment:
Implementing `ResetInterface` might be useful for testing, like https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Validator/Validator/TraceableValidator.php

Reply:
done!